n_queen_problem/NQueenProblem.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Implementation of popular N-Queen Problem using Backtracking algorithm.
+ * The size of a chess board is given, suppose n, then we have to suggest such a configuration than n queens could be placed on the chess board and no queen is under attack
+ * There can be a number of possible solutions for a specific board.
+ * This implementation prints only one valid configuration, it can be extended to print all possible valid configurations.
+ * A good example of recursion.
+ * You can see the solution for various sizes by changing the value of Size in 14th line..
+ */
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
```

Comment:
`memset` can be replaced with `std::fill` in `C++`.

Also it would be better if you use `std::vector` instead of 2D array for storing board as it will be very much easy to initialize it as compared to 2D array. @aviaryan : Views?